### DISCUSSION TITLE
How to enable Onyx lights driver?

### DISCUSSION THREAD

**Discussion by truekaerf** [koreader/koreader#8861] [Q&A]
I have an Onyx Boox Nova 3 Color with 2022.02 installed.  I saw in the release notes for 2022.01 that there was a new lights driver added for Onyx and specifically the 3 Color (#8562)  How do I enable that?  Going to the compatibility test screen, it shows EPD: qualcomm, Lights: generic.

**Comment by Frenzie** [comment]
You mean settings (gear) → frontlight? There shouldn't be anything to enable.

**Reply by truekaerf** [reply]
No.  I mean settings (gear) → frontlight still does nothing.  The way the patch notes state "New lights driver for Onyx", I was thinking that the test screen should indicate something other than generic.  Either way, I still can't adjust my frontlight from the app.

**Reply by Frenzie** [reply]
What else does it say in the test? As in https://github.com/koreader/koreader/issues/8482

**Reply by truekaerf** [reply]
Here is the test.log
[test.log](https://github.com/koreader/koreader/files/8156739/test.log)

**Reply by Frenzie** [reply]
It looks like the device wasn't included in LightsFactory.kt in <https://github.com/koreader/android-luajit-launcher/pull/345>, but I'm not sure if that was on purpose. Pinging @pazos

**Reply by truekaerf** [reply]
I see what you mean.  Good catch.

**Reply by pazos** [reply]
My magician hat doesn't work that way :p.

I asked the reporter about the light behaviour since he didn't give feedback. No feedback yet: https://github.com/koreader/koreader/issues/4551#issuecomment-983130485

Now I finally know it doesn't work with generic light driver but that doesn't solve the other question: is there a supported drivers that works well with that specific device. In that case which one?. I'm still assuming the device has no warmth.

**Reply by truekaerf** [reply]
The device only has light level, no warmth...in fact the system menu calls it "cold light".  I tried the other 3 drivers listed on the test screen but none of them had any effect.  If there's anything I can do on my end, I'm happy to try and help.
